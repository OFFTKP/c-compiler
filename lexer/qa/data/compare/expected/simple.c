int 19
main 60
( 59
int 19
argc 60
, 59
char 5
* 59
* 59
argv 60
) 59
{ 59
int 19
i 60
= 59
1000 35
; 59
i 60
++ 50
; 59
++ 50
i 60
; 59
i 60
-- 51
; 59
-- 51
i 60
; 59
i 60
>>= 38
1 35
; 59
i 60
<<= 39
2 35
; 59
i 60
+= 40
1 35
; 59
i 60
-= 41
1 35
; 59
if 17
( 59
i 60
== 57
1 35
|| 54
i 60
!= 58
2 35
) 59
return 23
0 35
; 59
if 17
( 59
i 60
>= 56
4 35
&& 53
i 60
<= 55
8 35
) 59
return 23
1 35
; 59
const 6
char 5
* 59
str 60
= 59
"this is a string//literal += 100||" 37
; 59
} 59
