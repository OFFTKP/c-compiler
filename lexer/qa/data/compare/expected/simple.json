{"Tokens":[{"Int":"int"},{"Identifier":"main"},{"Punctuator":"("},{"Int":"int"},{"Identifier":"argc"},{"Punctuator":","},{"Char":"char"},{"Punctuator":"*"},{"Punctuator":"*"},{"Identifier":"argv"},{"Punctuator":")"},{"Punctuator":"{"},{"Int":"int"},{"Identifier":"i"},{"Punctuator":"="},{"IntegerConstant":"1000"},{"Punctuator":";"},{"Identifier":"i"},{"IncOp":"++"},{"Punctuator":";"},{"IncOp":"++"},{"Identifier":"i"},{"Punctuator":";"},{"Identifier":"i"},{"DecOp":"--"},{"Punctuator":";"},{"DecOp":"--"},{"Identifier":"i"},{"Punctuator":";"},{"Identifier":"i"},{"RightAssign":">>="},{"IntegerConstant":"1"},{"Punctuator":";"},{"Identifier":"i"},{"LeftAssign":"<<="},{"IntegerConstant":"2"},{"Punctuator":";"},{"Identifier":"i"},{"AddAssign":"+="},{"IntegerConstant":"1"},{"Punctuator":";"},{"Identifier":"i"},{"SubAssign":"-="},{"IntegerConstant":"1"},{"Punctuator":";"},{"If":"if"},{"Punctuator":"("},{"Identifier":"i"},{"EqOp":"=="},{"IntegerConstant":"1"},{"OrOp":"||"},{"Identifier":"i"},{"NeOp":"!="},{"IntegerConstant":"2"},{"Punctuator":")"},{"Return":"return"},{"IntegerConstant":"0"},{"Punctuator":";"},{"If":"if"},{"Punctuator":"("},{"Identifier":"i"},{"GeOp":">="},{"IntegerConstant":"4"},{"AndOp":"&&"},{"Identifier":"i"},{"LeOp":"<="},{"IntegerConstant":"8"},{"Punctuator":")"},{"Return":"return"},{"IntegerConstant":"1"},{"Punctuator":";"},{"Const":"const"},{"Char":"char"},{"Punctuator":"*"},{"Identifier":"str"},{"Punctuator":"="},{"StringLiteral":"this is a string//literal += 100||"},{"Punctuator":";"},{"Punctuator":"}"},{"Eof":""}]}
